evalml/tests/component_tests/test_components.py

```diff
@@ -852,3 +853,57 @@ def test_component_equality_all_components(component_class):
     parameters = component.parameters
     equal_component = component_class(**parameters)
     assert component == equal_component
+
+
+def test_categorical_hyperparameters(X_y_binary):
```

Comment:
I think we can condense these two tests into one test with `pytest.mark.parametrize`. I also think there is value in testing that a component that that uses `Categorical` can be used in automl search since that is where they hyperparameter ranges are used.